source/interface/transport_interface.h

```diff
@@ -132,7 +147,14 @@
  *                          pBuffer,
  *                          bytesToSend,
  *                          MY_SOCKET_TIMEOUT );
- *     if( bytesSent < 0 )
+ *
+ *      // If underlying TCP buffer is full, set the return value to zero
+ *      // so that caller can retry operation.
+ *     if( bytesSent == TCP_BUFFER_FULL )
+ *     {
+ *          bytesSent = 0;
```

Comment:
Similar comment as per the receive - when the error is not really an error, rather just a full buffer, the error has to be translated into whatever the calling function interprets as meaning 'try again' until you time out.

Reply:
Yes, the zero return value represents that the operation can be re-tried.